pkg/controller/configmap/config.go

```diff
@@ -112,7 +66,38 @@ func validateConfig(cfg *ini.File, sec string, key string) error {
 	return nil
 }
 
-func validateConfigMap(configmap *corev1.ConfigMap) []error {
```

Comment:
Not related to your patch, but do you think we should move all functions in this file to ReconcileConfigMap class? Is that a better style in go?

Reply:
I only did the necessary moving (e.g. the logics for k8s and openshift are different)... Also not sure whether what you said is better.

Reply:
The function is already applied to ReconcileConfigMap type. Danting, do you mean doing the same for all functions?

Reply:
I think we don't need to move all functions in this file to ReconcileConfigMap class if not necessary. e.g. I saw https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/deployment/rollback.go didn't do that.